primitives/src/pallet_traits/pallet_staking_config.rs

```diff
@@ -0,0 +1,99 @@
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+*/
+use crate::FrameSystemConfig;
+use codec::MaxEncodedLen;
+use scale_info::TypeInfo;
+use sp_runtime::{
+	traits::{AtLeast32BitUnsigned, Get, MaybeSerializeDeserialize},
+	Perbill,
+};
+use sp_staking::{EraIndex, SessionIndex};
+
+/// The balance type of this pallet.
+pub type BalanceOf<T> = <T as StakingConfig>::CurrencyBalance;
+
+/// Simplifed pallet staking Config trait. Needed because substrate pallets compile to wasm
+/// in no_std mode.
+pub trait StakingConfig: FrameSystemConfig {
+	type Currency;
+	type CurrencyBalance: AtLeast32BitUnsigned
+		+ codec::FullCodec
+		+ Copy
+		+ MaybeSerializeDeserialize
+		+ core::fmt::Debug
+		+ Default
+		+ From<u64>
+		+ TypeInfo
+		+ MaxEncodedLen;
+	type UnixTime;
+	type CurrencyToVote;
+	type ElectionProvider;
+	type GenesisElectionProvider;
+	type MaxNominations: Get<u32>;
+	type HistoryDepth: Get<u32>;
+	type RewardRemainder;
+	type RuntimeEvent;
+	type Slash;
+	type Reward;
+	type SessionsPerEra: Get<SessionIndex>;
+	type BondingDuration: Get<EraIndex>;
+	type SlashDeferDuration: Get<EraIndex>;
+	type SlashCancelOrigin;
+	type SessionInterface;
+	type EraPayout;
+	type NextNewSession;
+	type MaxNominatorRewardedPerValidator: Get<u32>;
+	type OffendingValidatorsThreshold: Get<Perbill>;
+	type VoterList;
+	type TargetList;
+	type MaxUnlockingChunks: Get<u32>;
+	type OnStakerSlash: sp_staking::OnStakerSlash<Self::AccountId, BalanceOf<Self>>;
+	type WeightInfo;
+}
+
+#[cfg(feature = "std")]
+impl<T> StakingConfig for T
+where
+	T: pallet_staking::Config,
+{
+	type Currency = T::Currency;
+	type CurrencyBalance = T::CurrencyBalance;
+	type UnixTime = T::UnixTime;
+	type CurrencyToVote = T::CurrencyToVote;
+	type ElectionProvider = T::ElectionProvider;
+	type GenesisElectionProvider = T::GenesisElectionProvider;
+	type MaxNominations = T::MaxNominations;
+	type HistoryDepth = T::HistoryDepth;
+	type RewardRemainder = T::RewardRemainder;
+	type RuntimeEvent = <T as pallet_staking::Config>::RuntimeEvent;
+	type Slash = T::Slash;
+	type Reward = T::Reward;
+	type SessionsPerEra = T::SessionsPerEra;
+	type BondingDuration = T::BondingDuration;
+	type SlashDeferDuration = T::SlashDeferDuration;
+	type SlashCancelOrigin = T::SlashCancelOrigin;
+	type SessionInterface = T::SessionInterface;
+	type EraPayout = T::EraPayout;
+	type NextNewSession = T::NextNewSession;
+	type MaxNominatorRewardedPerValidator = T::MaxNominatorRewardedPerValidator;
+	type OffendingValidatorsThreshold = T::OffendingValidatorsThreshold;
+	type VoterList = T::VoterList;
+	type TargetList = T::TargetList;
+	type MaxUnlockingChunks = T::MaxUnlockingChunks;
+	type OnStakerSlash = T::OnStakerSlash;
+	type WeightInfo = T::WeightInfo;
+}
```

Comment:
So we don't offer yet an implementation for no-std for all those pallets? Well, to me this is fine.

Reply:
Not yet, at least. The whole api (except for utils) is not yet no_std compatible. So it just doesn't make sense, since the implementation which are now generic over the Runtime are located in the api / extrinsic.

That will be part of a future PR :)

Reply:
Alright, this makes sense!